### PR TITLE
Identify renaming a parameter set as non-breaking in the breaking change contract document

### PR DESCRIPTION
**Pull request by KirkMunro**
# PR Summary

Add a line that clearly identifies renaming a parameter set as a non-breaking change.

## PR Context

https://github.com/PowerShell/PowerShell/issues/10058

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by iSazonov** [comment]
I think it is not right change - parameter sets is public. See #10099

**Comment by KirkMunro** [comment]
Yet in https://github.com/PowerShell/PowerShell/issues/10058, @SteveL-MSFT says the following:

> @PowerShell/powershell-committee reviewed this. We reaffirm that **ParameterSet names are not a public api** and changing them is not a breaking change. We should update the breaking change contract doc to make this apparent and explicit.

Parameter set names are visible/accessible via cmdlet metadata and open source code, but they are not considered a public API that someone should take dependencies on.

**Review by iSazonov — DISMISSED** [review]

**Comment by iSazonov** [comment]
Perhaps it is my misunderstanding public api in code and in documentation.

**Comment by iSazonov** [comment]
We could add "System.Management.Automation.Internal" too. It was mentioned the other day.

**Comment by vexx32** [comment]
I think the distinction that is being drawn (and I agree, does need to be drawn) is:

1. Parameter sets themselves (their "shape" and contents, so to speak) are a public API (they affect how a cmdlet can be used, therefore removing / significant changes to an existing set can be breaking), _but_
2. The internally-recorded **name** for any given parameter set isn't something people should be relying upon.

I hope that makes it a bit more clear?

With regards to S.M.A.Internal, I'd agree it should probably be explicitly mentioned as some of the types and members there are public and usable from PS itself, so the nature of the contract in that area should be clarified. Not sure if @SteveL-MSFT wants to make a separate decision on appropriate wording / additions there. 🙂

**Comment by SteveL-MSFT** [comment]
The discussion in @PowerShell/powershell-committee was specifically about renaming a parameterset which was reaffirmed as not considered a breaking change.  The parameter sets themselves are considered a public API and those changes would be breaking.  As for SMA.Internal namespace, it is by definition not a public API even though can be called without reflection.  We don't document those APIs and users should not depend on them not changing.  Seems like `internal` shouldn't need to be called out explicitly.

File: docs/dev-process/breaking-change-contract.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ Changes to surface area or behavior that is clearly internal or non-breaking in
 Examples:
 
 + Changes to internal APIs that break private reflection
```
Comment (SteveL-MSFT) [inline docs/dev-process/breaking-change-contract.md]: If you want, you can append something like "including all APIs in System.Management.Automation.Internal namespace."  Not required for this PR.

**Review by SteveL-MSFT — APPROVED** [review]

File: docs/dev-process/breaking-change-contract.md (hunk 1 above)
Comment (KirkMunro) [reply docs/dev-process/breaking-change-contract.md]: Added note.

**Review by iSazonov — APPROVED** [review]

File: docs/dev-process/breaking-change-contract.md [hunk 2]
```diff
@@ -71,7 +71,9 @@ Changes to surface area or behavior that is clearly internal or non-breaking in
 
 Examples:
 
-+ Changes to internal APIs that break private reflection
++ Changes to internal APIs that break private reflection.
++ Changes to APIs in the System.Management.Automation.Internal namespace (even if they are public, they are still considered internal and subject to change).
```
Comment (iSazonov) [inline docs/dev-process/breaking-change-contract.md]: To fix Static analysis CI:
```suggestion
+ Changes to APIs in the `System.Management.Automation.Internal` namespace (even if they are public, they are still considered internal and subject to change).
```

**Pull request merged**